senator/game.py

```diff
@@ -102,21 +109,29 @@ def solve_node_via_rm_plus(self, node: Node, iterations: int = 1000) -> None:
         if (conv := node.nash_conv()) >= 0.001:
             print(f"Nash Convexity is {conv}, something is wrong...")
 
-    def run_greedy_search(self, num_expansions: int = 10):
-        """
-        Run a greedy search algorithm, periodically expanding the tree. Each expansion is done by first solving the
-        node using the RM+ algorithm, sampling a joing action, and adding the child to the tree.
-        TODO: decide how to do the expanding. Baseline is to just do a path of length 5.
-        """
+    def run_greedy_search(self, num_expansions: int = 17) -> np.ndarray:
+        
+        actions_in_steps = []
+        strategies_in_steps = []
+        
         current_node = self.root
         utility = self.initial_utility
-        for _ in range(num_expansions):
+
+        print(f"Initial utility\nu: {utility.sum(axis=1)}")
+
+        for i in range(num_expansions):
```

Comment:
Tady děláš `for i in ...`, ale pak to `i` nikde nepoužíváš (pokud nejsem slepý). V takovém případě je lepší udělat `for _ in ...`, čímž dáváš na jevo že skutečně to `i` nebudeš potřebovat.

Reply:
Yes, tohle je nějaký zmetek po předchozí iteraci kódu, kde jsem přes ty i-čka vstupoval do pole. Pak jsem přišel na append, a tohle už neupravil. Upraveno.

Reply:
Počkat, já tam dole printím číslo dne, kde to i používám. Takže nechávám i.

Reply:
Jo sorry